Cabal/Distribution/Types/VersionRange.hs

```diff
@@ -469,6 +479,31 @@ instance Parsec VersionRange where
                 (orLaterVersion u) (earlierVersion (majorUpperBound u))
             embed vr = embedVersionRange vr
 
+        -- version set notation (e.g. "== { 0.0.1.0, 0.0.2.0, 0.1.0.0 }")
+        verSet' op vs = do
+            csv <- askCabalSpecVersion
+            if csv >= CabalSpecV3_0
+            then pure $ foldr1 unionVersionRanges (map op vs)
+            else fail $ unwords
+                [ "version set syntax used."
+                , "To use this syntax the package needs to specify at least 'cabal-version: 3.0'."
+                , "Alternatively, if broader compatibility is important then use"
+                , "a series of single version constraints joined with the || operator:"
+                , prettyShow (foldr1 unionVersionRanges (map op vs))
+                ]
+
+        verSet :: CabalParsing m => m [Version]
+        verSet = do
+            _ <- P.char '{'
+            vs <- P.sepBy1 (P.spaces *> verPlain) (P.try (P.spaces *> P.char ','))
```

Comment:
this can be rewritten without `try`, if you eat spaces after the parsers here.

```
_ <- P.char '{'
P.spaces
vs <- P.sepBy (verPlain <* spaces) (P.char ',' *> P.spaces)
_ <- P.char '}'
```

Reply:
small typo though, should have read

```
vs <- P.sepBy1 (verPlain <* P.spaces) (P.char ',' *> P.spaces)
```